consensus/istanbul/validator/weighted.go

```diff
@@ -290,97 +296,82 @@ func (valSet *weightedCouncil) List() []istanbul.Validator {
 	valSet.validatorMu.RLock()
 	defer valSet.validatorMu.RUnlock()
-	if uint64(len(valSet.validators)) <= valSet.subSize {
-		logger.Trace("Council is less than subsize Choose all validators", "valSet.Number", valSet.blockNum, "prevHash", prevHash.Hex(), "proposer", proposer, "committee", valSet.validators, "committee size", len(valSet.validators), "subsize", valSet.subSize)
-		return valSet.validators
-	}
 
-	hashstring := strings.TrimPrefix(prevHash.Hex(), "0x")
-	if len(hashstring) > 15 {
-		hashstring = hashstring[:15]
-	}
+	validators := valSet.validators
+	validatorSize := uint64(len(validators))
+	committeeSize := valSet.subSize
 
-	seed, err := strconv.ParseInt(hashstring, 16, 64)
-	if err != nil {
-		logger.Error("input", "hash", prevHash.Hex())
-		logger.Error("fail to make sub-list of validators", "seed", seed, "err", err)
-		return valSet.validators
+	// find the proposer
+	proposerIdx, proposer := valSet.getByAddress(proposerAddr)
+	if proposerIdx < 0 {
+		logger.Error("invalid index of the proposer",
+			"addr", proposerAddr.String(), "index", proposerIdx)
+		return validators
 	}
 
-	// shuffle
-	committee := make([]istanbul.Validator, valSet.subSize)
-	_, proposerValidator := valSet.getByAddress(proposer)
-	if proposerValidator == nil {
-		logger.Error("fail to make sub-list of validators, because proposer is invalid", "address of proposer", proposer)
-		return valSet.validators
-	}
-	committee[0] = proposerValidator
-	if valSet.subSize == 1 {
-		logger.Debug("Sub size is 1, current", "prevHash", prevHash.Hex(), "proposer", proposer, "committee", committee, "committee size", len(committee), "subSize", valSet.subSize)
-		return committee
-	}
-	// next proposer
-	nextProposerIdx := uint64(1)
-	for committee[1] == nil {
-		nextProposer := valSet.selector(valSet, proposer, view.Round.Uint64()+nextProposerIdx)
-		if committee[0].Address() != nextProposer.Address() {
-			committee[1] = nextProposer
-			break
-		}
-		nextProposerIdx += 1
-	}
-
-	proposerIdx, _ := valSet.getByAddress(committee[0].Address())
-	nextproposerIdx, _ := valSet.getByAddress(committee[1].Address())
-
-	// TODO-Klaytn-RemoveLater remove this check code if the implementation is stable.
-	if proposerIdx < 0 || nextproposerIdx < 0 {
-		vals := "["
-		for _, v := range valSet.validators {
-			vals += fmt.Sprintf("%s,", v.Address().Hex())
-		}
-		vals += "]"
-		logger.Error("current proposer or next proposer not found in Council", "proposerIdx", proposerIdx, "nextproposerIdx", nextproposerIdx, "proposer", committee[0].Address().Hex(),
-			"nextproposer", committee[1].Address().Hex(), "validators", vals)
+	// return early if the committee size is an edge value
+	if committeeSize == 1 {
+		return []istanbul.Validator{proposer}
 	}
-
-	if proposerIdx == nextproposerIdx {
-		logger.Error("fail to make propser", "current proposer idx", proposerIdx, "next idx", nextproposerIdx)
+	if committeeSize >= validatorSize {
+		return validators
 	}
```

Comment:
how about moving these codes to top of the function.
we can just return if the commiteeSize is bigger than validatorSize

Reply:
Thank you. I moved it!